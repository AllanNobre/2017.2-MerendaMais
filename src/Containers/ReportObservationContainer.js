import { connect } from 'react-redux';
import ReportObservationScreen from '../screens/ReportObservationScreen';
import { setReportObservation } from '../actions/reportActions';

const mapStateToProps = state => ({
  observation: state.report.otherObservation,
});

const mapDispatchToProps = dispatch => ({
  setReportObservation: observation => dispatch(setReportObservation(observation)),
});

const ReportObservationContainer =
connect(mapStateToProps, mapDispatchToProps)(ReportObservationScreen);

export default ReportObservationContainer;
