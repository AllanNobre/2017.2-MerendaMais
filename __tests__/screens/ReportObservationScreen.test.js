import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// imported as a connected component!
import ReportObservationContainer from '../../src/Containers/reports/ReportObservationContainer';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  report: {
    reportObservation: '',
  },
};

const store = mockStore(initialState);

describe('Testing ReportObservation Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <ReportObservationContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
