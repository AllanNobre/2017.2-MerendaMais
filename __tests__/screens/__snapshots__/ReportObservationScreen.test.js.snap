// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReportObservation Screen renders as expected 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "white",
      "flex": 6,
      "flexDirection": "column",
      "marginBottom": 9,
    }
  }
>
  <Header
    backButton={true}
    subTitle="Observações adicionais"
    title="Relatório"
  />
  <View
    behavior="padding"
  >
    <View
      style={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 30,
        }
      }
    >
      <TextInput
        multiline={true}
        onChangeText={[Function]}
        placeholder="Observações gerais que gostaria de adicionar ?(opcional)"
        style={
          Object {
            "borderRadius": 12,
            "borderWidth": 1,
            "fontSize": 37.5,
            "height": 333.5,
            "paddingLeft": 10,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    key="setObservation"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#FF9500",
        "borderRadius": 7,
        "borderWidth": 1,
        "justifyContent": "flex-end",
        "marginBottom": 20,
        "marginHorizontal": 15,
        "marginTop": 30,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "textAlign": "center",
        }
      }
    >
      Concluir
    </Text>
  </TouchableOpacity>
</ScrollViewMock>
`;
